in_toto/model_test.go

```diff
@@ -1128,3 +1128,54 @@ func TestValidateSupplyChainItem(t *testing.T) {
 		}
 	}
 }
+
+func TestMetablockSignWithEd25519(t *testing.T) {
+	// Test metablock signing (with ed25519)
+	// - Pass non-private key
+	// - Pass unsupported invalid key type
+	// - Pass an ed25519 key and expect a signature back
+	var key Key
+	var mb Metablock
+	mb.Load("demo.layout.template")
+
+	key.LoadPublicKey("alice.pub")
+	err := mb.Sign(key)
+	if err == nil || !strings.Contains(err.Error(), "supported yet") {
+		t.Errorf("Metablock.Sign returned (%s), expected it to claim this "+
+			"key type/scheme is unsupported", err)
+
+	}
+
+	validKey := `{"keytype": "ed25519", "scheme": "ed25519", "keyid": "308e3f53523b632983a988b72a2e39c85fe8fc967116043ce51fa8d92a6aef64", "keyid_hash_algorithms": ["sha256", "sha512"], "keyval": {"public": "8f93f549eb4cca8dc2142fb655ba2d0955d1824f79474f354e38d6a359e9d440", "private": "861fd1b466cfc6f73f8ed630f99d8eda250421f0e3a6123fd5c311cc001bda49"}}`
+
+	badkey, err := ParseEd25519FromPrivateJSON(validKey)
+	if err != nil {
+		t.Errorf("ParseEd25519FromPrivateJSON returned (%s), expected no error",
+			err)
+	}
+
+	badkey.Scheme = "ecdsa"
+	err = mb.Sign(badkey)
+	if err == nil || !strings.Contains(err.Error(), "not supported") {
+		t.Errorf("Metablock.Sign returned (%s), expected it to claim this "+
+			"key type/scheme is unsupported", err)
+	}
+
+	key, err = ParseEd25519FromPrivateJSON(validKey)
+	if err != nil {
+		t.Errorf("ParseEd25519FromPrivateJSON returned (%s), expected no error",
+			err)
+	}
+
+	err = mb.Sign(key)
+	if err != nil {
+		t.Errorf("Metablock.Sign returned (%s), expected no error", err)
+	}
+
+	// note, there's a 2 because this template is already signed hehe
+	if len(mb.Signatures) != 2 {
+		t.Errorf("Expected a new signature to be appended, but got (%+v) (%d)",
+			mb.Signatures, len(mb.Signatures))
+	}
```

Comment:
Have you tried verifying the signature? Or do you want to do this in a separate PR?

Reply:
I want to abstract verification functions a little bit in a separate PR, if that's ok with you :)